File: backend_healthconnect/src/main/java/com/backend_healthconnect/model/metricheSaluteDTO.java
package com.backend_healthconnect.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class metricheSaluteDTO {
    private Long id;
    private utenteDTO paziente;
    private utenteDTO medico;
    private TipoMetrica tipoMetrica;
    private Double valore;
    private String unità_misura;
    private LocalDate data;
}

File: frontend_healthconnect/src/app/model/metricheSaluteDTO.ts
import { utenteDTO } from './utenteDTO';

export interface MetricheSaluteDTO {
  id: number;
  paziente: utenteDTO;
  medico: utenteDTO;
  tipoMetrica: TipoMetrica;
  valore: number;
  unità_misura: string;
  data: string;
}

export enum TipoMetrica {
  PESO = 'PESO',
  PRESSIONE_MAX = 'PRESSIONE_MAX',
  PRESSIONE_MIN = 'PRESSIONE_MIN',
  GLICEMIA = 'GLICEMIA'
}

